sys-boot/udk/udk-2018.ebuild

```diff
@@ -0,0 +1,256 @@
+	if tc-is-gcc; then
+		if ver_test $(gcc-version) -ge 5; then
+			TOOLCHAIN_TAG="GCC5"
+		elif ver_test $(gcc-version) -ge 4.9; then
+			TOOLCHAIN_TAG="GCC49"
+		elif ver_test $(gcc-version) -ge 4.8; then
+			TOOLCHAIN_TAG="GCC48"
+		elif ver_test $(gcc-version) -ge 4.7; then
+			TOOLCHAIN_TAG="GCC47"
+		elif ver_test $(gcc-version) -ge 4.6; then
+			TOOLCHAIN_TAG="GCC46"
+		elif ver_test $(gcc-version) -ge 4.5; then
+			TOOLCHAIN_TAG="GCC45"
+		elif ver_test $(gcc-version) -ge 4.4; then
+			TOOLCHAIN_TAG="GCC44"
+		fi
+	elif tc-is-clang; then
+		if ver_test $(clang-version) -ge 3.8; then
+			TOOLCHAIN_TAG="CLANG38"
+		elif ver_test $(clang-version) -ge 3.5; then
+			TOOLCHAIN_TAG="CLANG35"
+		fi
+	fi
+	if [[ -z ${TOOLCHAIN_TAG} ]]; then
+		TOOLCHAIN_TAG="ELFGCC"
+	fi
+}
+
+src_unpack() {
+	unpack ${A}
+
+	local doc_name
+	local f
+	if use doc; then
+		rm -f "${DOCDIR}/"*.chm || die
+		pushd "${DOCDIR}" >/dev/null || die
+		for f in "${DOCDIR}/"*.zip; do
+			doc_name=$(basename "${f}" .zip)
+			unpack "${f}"
+			mv "${DOCDIR}/html" "${DOCDIR}/${doc_name}" || die
+		done
+		popd >/dev/null || die
+		rm -f "${DOCDIR}/"*.zip || die
+	fi
+}
+
+src_configure() {
+	# Compile of Base Tools is required for further setting up the environment
+	# Base tools does not like parallel make
+	local cflags_save=${CFLAGS}
+	append-cflags -MD -fshort-wchar -fno-strict-aliasing
+	append-cflags -nostdlib -c -fPIC
+	sed -e "s:^\(BUILD_CFLAGS\s*=\).*$:\1 ${CFLAGS}:" \
+		-i "${S}/BaseTools/Source/C/Makefiles/header.makefile" \
+		|| die "Failed to update makefile header"
+	CFLAGS=${cflags_save}
+	local make_flags=(
+		BUILD_CC="$(tc-getBUILD_CC)"
+		BUILD_CXX="$(tc-getBUILD_CXX)"
+		BUILD_AS="$(tc-getBUILD_AS)"
+		BUILD_AR="$(tc-getBUILD_AR)"
+		BUILD_LD="$(tc-getBUILD_LD)"
+	)
+	emake "${make_flags[@]}" -j1 -C BaseTools
+	. edksetup.sh
+
+	# Update UDK parameter files
+	sed -e "s:^\(ACTIVE_PLATFORM\s*=\).*$:\1 MdeModulePkg/MdeModulePkg.dsc:" \
+		-e "s:^\(TARGET\s*=\).*$:\1 RELEASE:" \
+		-e "s:^\(TARGET_ARCH\s*=\).*$:\1 ${ARCH}:" \
+		-e "s:^\(TOOL_CHAIN_TAG\s*=\).*$:\1 ${TOOLCHAIN_TAG}:" \
+		-e "s:^\(MAX_CONCURRENT_THREAD_NUMBER\s*=\).*$:\1 $(makeopts_jobs):" \
+		-i "${S}/Conf/target.txt" || die "Failed to configure target file"
+	sed -e "s:\(_CC_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASM_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_PP_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_VFRPP_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASLCC_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASLPP_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_DLINK_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASLDLINK_PATH\s*=\).*$:\1 $(tc-getLD):" \
+		-e "s:\(_OBJCOPY_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+		-e "s:\(_RC_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+		-e "s:\(_SLINK_PATH\s*=\).*$:\1 $(tc-getAR):" \
+		-i "${S}/Conf/tools_def.txt" \
+		|| die "Failed to prepare tools definition file"
+}
+
+src_compile() {
+	local build_target
+	if use examples; then
```

Comment:
USE=examples is about installing source code of examples (for user to read), not building them.

Reply:
Actually, we need to build all in order for examples to be generated, and not only compiled. Some files do not exist in sources and are generated from template and depending on chosen toolchain.